clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -296,9 +281,11 @@ public boolean awaitPendingRequests(Node node, long timeoutMs) {
      */
     public int pendingRequestCount(Node node) {
         synchronized (this) {
-            List<ClientRequest> pending = unsent.get(node);
-            int unsentCount = pending == null ? 0 : pending.size();
-            return unsentCount + client.inFlightRequestCount(node.idString());
+            int total = 0;
+            for (UnsentRequest request : unsent)
```

Comment:
One thing we can do to reduce the cost of this is to add `hasPendingRequests` methods to `ConsumerNetworkClient`. Most of the uses don't actually need the count and this lets us shortcut. Note also that `size()` is O(n) for `ConcurrentLinkedQueue`, so being able to use `isEmpty()` would be a nice improvement.